rocksdb_admin/admin_handler.cpp

```diff
@@ -1070,7 +1081,16 @@ void AdminHandler::async_tm_startMessageIngestion(
 
     // TODO: add the logic to write the key/value to rocksdb here
 
-    // TODO: periodically write the message timestamp to local_meta_db
+    if (!is_replay && (message_count % message_interval == 0)) {
```

Comment:
what does is_replay exactly mean and why we don't want to write timestamp if is_replay is true?

Reply:
is_replay flag is set to true when the messages being consumed are part of the catch up (and not live messages). Two reasons:
1. We want this to be as performant as possible since this part is blocking the transition to the online phase. With a message interval set to 1000, we would end up writing to meta_db multiple times a second (per shard). This is not necessary.
2. The periodic update is targeted towards update/restart cases. And the likelihood of that happening when a new batch is uploaded or when a shard is introduced to a new host is less.

Reply:
Thanks for the information. I am just a little concerned that how reliable is this is_replay flag and how its value is determined? Will it be true if there is a little network glitch or slowness in the kafka consumer side? If is_replay happens to be set to true for the message whose message_count is % message_interval == 0, we may skip some timestamp persistence and get some non-deterministic effect.

The overhead of one extra read and one extra write to meta_db for every 1000 DB writes seems to be small to me.

Reply:
If we are sure that is_replay is reliable, I am ok with it. just let me know.

Reply:
is_replay is set to true for the synchronous consumption. Once we start consuming live messages, a new thread is spawned by the watcher and all further messages will have the flag set to false (irrespective of network or kafka delays). 
But I don't have a strong opinion for/against this. I'll remove it, we can reassess it later if this does affect performance.